packages/application-components/src/experimental/hooks/use-debounced-promise-callback/use-debounced-promise-callback.ts

```diff
@@ -0,0 +1,9 @@
+import debounce from 'debounce-promise';
+import { useCallback } from 'react';
+
+const useDebouncedPromiseCallback = <T extends (...args: any[]) => any>(
+  promiseCallback: T,
+  wait: number
+) => useCallback(debounce(promiseCallback, wait), [promiseCallback]);
```

Comment:
Shouldn't this be something like:

```suggestion
const useDebouncedPromiseCallback = <T>(
  promiseCallback: (...args: unknown[]) => Promise<T>,
  wait: number
) => useCallback(debounce(promiseCallback, wait), [promiseCallback]);
```